tensilelite/script/retune.py

```diff
@@ -0,0 +1,185 @@
+        isLSUenable = True
+        newMi8 = mi8 // 2
+        newMi6 = mi6 * 2
+        newMi = copy.deepcopy(baseMi[0][0])
+        newMi['WorkGroup'] = [16, 8, 2] #LSU2
+        newMi['MatrixInstruction'][6] = newMi6 #miwt0
+        newMi['MatrixInstruction'][8] = newMi8 #miwt0
+        baseMi[0].append(newMi)
+    if (mi8%2) == 0 and (mi7%2) == 0:
+        #Generate a LSU4 MatrixInstruction
+        isLSUenable = True
+        newMi8 = mi8 // 2
+        newMi6 = mi6 * 2
+        newMi = copy.deepcopy(baseMi[0][0])
+        newMi['WorkGroup'] = [16, 8, 4] #LSU2
+        newMi['MatrixInstruction'][6] = newMi6 #miwt0
+        newMi['MatrixInstruction'][8] = newMi8 #miwt0
+        newMi7 = mi7 // 2
+        newMi5 = mi5 * 2
+        newMi['MatrixInstruction'][5] = newMi5 #miwt0
+        newMi['MatrixInstruction'][7] = newMi7 #miwt0
+        baseMi[0].append(newMi)
+    print(baseMi)
+    print(MT)
+    print("======END=====")
+
+    with open("tune.yaml") as stream:
+    #with open("../HHS_TN_LLAMA_N1_01_Msmall_v3.yaml") as stream:
+        data_loaded = yaml.safe_load(stream)
+        newPara = \
+        {'Groups': baseMi}
+        # {'Groups': 
+        # [[{'MatrixInstruction': [16, 16, 16, 1, 1, 1, 1, 1, 1], 'WorkGroup': [16, 4, 4], 'VectorWidthA': -1}, \
+        # {'MatrixInstruction': [16, 16, 16, 1, 1, 2, 1, 1, 1], 'WorkGroup': [16, 4, 4], 'VectorWidthA': 2}, \
+        # {'MatrixInstruction': [16, 16, 16, 1, 1, 3, 1, 1, 1], 'WorkGroup': [16, 4, 4], 'VectorWidthA': 1}, \
+        # {'MatrixInstruction': [16, 16, 16, 1, 1, 4, 1, 1, 1], 'WorkGroup': [16, 4, 4]}, \
+        # {'MatrixInstruction': [16, 16, 16, 1, 1, 1, 1, 2, 1], 'WorkGroup': [16, 8, 2], 'VectorWidthA': 1}, \
+        # {'MatrixInstruction': [16, 16, 16, 1, 1, 2, 1, 2, 1], 'WorkGroup': [16, 8, 2], 'VectorWidthA': 2}, \
+        # {'MatrixInstruction': [16, 16, 16, 1, 1, 1, 1, 4, 1], 'WorkGroup': [16, 16, 1], 'VectorWidthA': 1}]]} \
+
+        data_loaded["BenchmarkProblems"][0][1]['ForkParameters'].append(newPara)
+
+        try:
+            print(data_loaded["BenchmarkProblems"][0][1]['BenchmarkFinalParameters'][0]['ProblemSizes'][0])
+            data_loaded["BenchmarkProblems"][0][1]['BenchmarkFinalParameters'][0]['ProblemSizes'][0]['Exact'][0] = int(fullSize[0])
+            data_loaded["BenchmarkProblems"][0][1]['BenchmarkFinalParameters'][0]['ProblemSizes'][0]['Exact'][1] = int(fullSize[1])
+            data_loaded["BenchmarkProblems"][0][1]['BenchmarkFinalParameters'][0]['ProblemSizes'][0]['Exact'][2] = int(fullSize[2])
+            data_loaded["BenchmarkProblems"][0][1]['BenchmarkFinalParameters'][0]['ProblemSizes'][0]['Exact'][3] = int(fullSize[3])
+            print(data_loaded["BenchmarkProblems"][0][1]['BenchmarkFinalParameters'][0]['ProblemSizes'][0])
+        except yaml.YAMLError as exc:
+            print(exc)
+
+        if isLSUenable == True:
+            buildFolder = "/home/menghung/hcman/hipBLASLt/tensilelite/script/build"
```

Comment:
Don't use this kind of folder name

Reply:
accidentally pushed unnecessary data. Removed.